app/controllers/glossary_terms_controller.rb

```diff
@@ -99,28 +98,55 @@ def show_past
 
   # --------- Other private methods
 
-  def image_args
-    Rails.env.test? ? permit_upload_image_param : strong_upload_image_param
+  def init_image_form_instance_variables
+    @copyright_holder ||= @user.name
+    @copyright_year ||= Time.now.utc.year
+    @upload_license_id ||= @user.license_id
+    # rubocop:disable Naming/MemoizedInstanceVariableName
+    # RuboCop wants to name the ivar "@update_image_form_instance_variables"
+    @licenses ||= License.current_names_and_ids(@user.license)
+    # rubocop:enable Naming/MemoizedInstanceVariableName
   end
 
-  def strong_upload_image_param
-    {
-      copyright_holder: params[:copyright_holder],
-      when: Time.local(params[:date][:copyright_year]).utc,
-      license: License.safe_find(params[:upload][:license_id]),
-      user: @user,
-      image: params[:glossary_term][:upload_image]
-    }
+  def reload_form(form)
+    add_glossary_term_error_messages_to_flash
+    init_image_form_instance_variables
+    render(form)
   end
 
-  def permit_upload_image_param
-    args = strong_upload_image_param
-    # Remove "permitted: false" from this param so that model can mass assign it
-    args[:image] = params[:glossary_term][:upload_image][:image]
-    args
+  def add_glossary_term_error_messages_to_flash
+    @glossary_term.errors.messages.each_value do |val|
+      # flash_error takes a string; val is an array of size 1, e.g. ["message"]
+      flash_error(val.first)
+    end
+  end
```

Comment:
### Is there a better way to handle validation?

I want to:
- Have the model (not the controller) do the validation as closely as possible to the "Rails way". See https://guides.rubyonrails.org/v5.2/active_record_validations.html. It seems to me that this is "business logic" that should be handled by the model.
- Have the model provide an error message (sensu lato). Seems to me again to be business logic, though not as clear as the above.
- Have the error message be translatable by MO;
- Display the error message in the view.

My solution:
- GlossaryTerm uses `validates presence`, etc. See `glossary_terms.rb`
- GlossaryTerm provides an error message using a proc. Ex: `message: proc { :glossary_error_name_blank.t }` 
- Controller finds any error messages and puts them into the flash. I'm not in love with this, but I think it's required by the application layout. I think that normally, the view would handle this without intervention by the controller.

Reply:
Sounds exactly right to me.  I think we should follow that pattern everywhere.